### PR TITLE
PXS-943 Optimize tours

### PR DESCRIPTION
**Pull request by ToufiPF**
Optimize MapboxActivity:
- Clear Activity backstack when launching Mapbox to free memory + Launch FeedbackActivity onBackPressed
- Update Mapbox + few other dependencies
- Limit framerate to 30 fps
- Open POIs in a dedicated Activity => Mapbox activity is stopped automatically when a POI is displayed.
- Remove useless queryFeature in onMapClick (can work with Tour's POIs directly)

Fix bug in feedback activity:
- No longer possible to go back to FeedbackActivity by pressing back when on HomeActivity (FeedbackActivity clears activity stack on exit)

Design changes: POI is now displayed in an activity.
Exit button hovers over the content, and I added a small padding on top of the content so that the title is not hidden by the button.
<p align="left">

</p>

### CODE REVIEW

**Comment by timotheeduran** [comment]
Option 2 (Exit button hovers on top of content) is better.

**Comment by timotheeduran** [comment]
You can keep the shadow.

File: app/src/main/java/com/pindex/main/ui/activities/MapboxActivity.kt [hunk 1]
```diff
@@ -82,26 +75,38 @@ class MapboxActivity : AppCompatActivity(), SupportFullScreenGalleryActivity {
...
-                val dist = LocationUtils.getDistance(featurePoint.toGeoPoint(), p.geopoint!!)
-                if (dist < min) {
-                    min = dist
-                    closestPoint = p
+            var min = MIN_DISTANCE_FROM_CLICK
+
+            poiList?.forEach { poi ->
+                poi.geopoint?.let { geoPoint ->
+                    val dist = LocationUtils.getDistance(clickedPoint, geoPoint)
+                    println("ON MAP CLICK : distance $dist")
```
Comment (qtipee) [inline app/src/main/java/com/pindex/main/ui/activities/MapboxActivity.kt]: Remove `println` ?

File: app/src/main/java/com/pindex/main/ui/activities/MapboxActivity.kt [hunk 2]
```diff
@@ -127,19 +132,20 @@ class MapboxActivity : AppCompatActivity(), SupportFullScreenGalleryActivity {
         override fun onRotate(detector: RotateGestureDetector) {}
         override fun onRotateEnd(detector: RotateGestureDetector) {}
 
+        /**
+         * Get the Point of interest where user clicked, if any, and
+         * launches a [PoiContentActivity].
+         */
         override fun onMapClick(point: Point): Boolean {
-            val pixel = mapboxMap.pixelForCoordinate(point)
+            println("ON MAP CLICK : point $point ; ${tourDto.points?.size}")
```
Comment (qtipee) [inline app/src/main/java/com/pindex/main/ui/activities/MapboxActivity.kt]: Remove `println` ?

File: app/src/main/java/com/pindex/main/ui/activities/MapboxActivity.kt [hunk 3]
```diff
@@ -127,19 +132,20 @@ class MapboxActivity : AppCompatActivity(), SupportFullScreenGalleryActivity {
...
+            val poi = closestPoi(point.toGeoPoint(), tourDto.points) ?: return false
+
+            println("ON MAP CLICK : poi $poi")
```
Comment (qtipee) [inline app/src/main/java/com/pindex/main/ui/activities/MapboxActivity.kt]: Remove `println` ?

File: app/src/main/java/com/pindex/main/ui/activities/MapboxActivity.kt [hunk 4]
```diff
@@ -229,6 +213,7 @@ class MapboxActivity : AppCompatActivity(), SupportFullScreenGalleryActivity {
         initGestures()
         initLocationComponentAndCamera()
 
+        mapboxView.setMaximumFps(30)
```
Comment (qtipee) [inline app/src/main/java/com/pindex/main/ui/activities/MapboxActivity.kt:212]: Can we use a constant here instead of a magic number ?

**Review by qtipee — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/pindex/main/ui/activities/MapboxActivity.kt (hunk 1 above)
Comment (ToufiPF) [reply app/src/main/java/com/pindex/main/ui/activities/MapboxActivity.kt]: Yep done in the last commit

File: app/src/main/java/com/pindex/main/ui/activities/MapboxActivity.kt (hunk 4 above)
Comment (ToufiPF) [reply app/src/main/java/com/pindex/main/ui/activities/MapboxActivity.kt:212]: I'm on it

**Review by qtipee — APPROVED** [review]

**Pull request merged**